src/stdlib_math.fypp

```diff
@@ -1,21 +1,270 @@
 #:include "common.fypp"
 #:set IR_KINDS_TYPES = INT_KINDS_TYPES + REAL_KINDS_TYPES
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
 
 module stdlib_math
     use stdlib_kinds, only: int8, int16, int32, int64, sp, dp, qp
 
     implicit none
     private
-    public :: clip
+    public :: clip, linspace, logspace
+    public :: EULERS_NUMBER_SP, EULERS_NUMBER_DP, EULERS_NUMBER_QP
+    public :: DEFAULT_LINSPACE_LENGTH, DEFAULT_LOGSPACE_BASE, DEFAULT_LOGSPACE_LENGTH
+
+    integer, parameter :: DEFAULT_LINSPACE_LENGTH = 100
+    integer, parameter :: DEFAULT_LOGSPACE_LENGTH = 50
```

Comment:
Why is the lenght not 100, like for `default_linspace_lenght`?

Reply:
I chose the length to be 50 in accordance with `MATLAB`'s and `numpy`'s APIs

Reply:
`numpy` uses 50 for both `linspace` and logpspace`. Matlab uses 100 for linspace and 50 for logspace.

I think it would make sense to keep the same lenght for both implementations in a same module/library.

Reply:
I was unaware that `numpy` used 50 for `linspace` as well. I guess I am a creature of habit and `MATLAB` sank its hooks into me early :p. I will mark the conversation as unresolved and unless other people chime in to change it, I will leave it as be. FYI I know English isn't your first language but it's spelled leng**TH**. English is so nasty sometimes and even I misspell shit all the time.

Reply:
> FYI I know English isn't your first language but it's spelled leng**TH**. English is so nasty sometimes and even I misspell shit all the time.

Thank you for the note. Indeed, I should be more careful when writing.

Reply:
No worries 😃! As a language learner myself I know how easy it is to make mistakes when not paying attention.